File: conformance/tests/stateful/stateful.runar.py
from runar import StatefulSmartContract, Bigint, Readonly, public, assert_

class Stateful(StatefulSmartContract):
    count: Bigint
    max_count: Readonly[Bigint]

    def __init__(self, count: Bigint, max_count: Bigint):
        super().__init__(count, max_count)
        self.count = count
        self.max_count = max_count

    @public
    def increment(self, amount: Bigint):
        self.count = self.count + amount
        assert_(self.count <= self.max_count)

    @public
    def reset(self):
        self.count = 0

File: conformance/tests/stateful/stateful.runar.rb
require 'runar'

class Stateful < Runar::StatefulSmartContract
  prop :count, Bigint
  prop :max_count, Bigint, readonly: true

  def initialize(count, max_count)
    super(count, max_count)
    @count = count
    @max_count = max_count
  end

  runar_public amount: Bigint
  def increment(amount)
    @count = @count + amount
    assert @count <= @max_count
  end

  runar_public
  def reset
    @count = 0
  end
end
